### PR TITLE
Set a meta link for pr_build_link to pr_path

### PR DESCRIPTION
**Pull request by krzyzacy**
Prep for render pr-logs in testgrid

/assign @rmmh

### CODE REVIEW

File: jenkins/bootstrap.py [hunk 1]
```diff
@@ -284,6 +284,7 @@ def start(gsutil, paths, stamp, node_name, version, repos):
     # Upload a link to the build path in the directory
     if paths.pr_build_link:
         gsutil.upload_text(paths.pr_build_link, paths.pr_path)
```
Comment (rmmh) [inline jenkins/bootstrap.py]: Add a keyword argument to upload_text for optional headers instead-- this is a generally useful utility, and it wouldn't require two invocations.

Comment (rmmh) [inline jenkins/bootstrap.py]: i.e. `gsutil.upload_text(paths.pr_build_link, paths.pr_path, {'x-goog-meta-link': paths.pr_path})`

File: jenkins/bootstrap.py [hunk 2]
```diff
@@ -320,14 +324,18 @@ def copy_file(self, dest, orig):
         cmd = [self.gsutil, '-q', 'cp', '-Z', orig, dest]
         self.call(cmd)
 
-    def upload_text(self, path, txt, cached=True):
+    def upload_text(self, path, txt, meta_links={}, cached=True):
```
Comment (rmmh) [inline jenkins/bootstrap.py]: kwarg should be 'headers'

File: jenkins/bootstrap.py [hunk 3]
```diff
@@ -283,7 +283,11 @@ def start(gsutil, paths, stamp, node_name, version, repos):
     gsutil.upload_json(paths.started, data)
     # Upload a link to the build path in the directory
     if paths.pr_build_link:
-        gsutil.upload_text(paths.pr_build_link, paths.pr_path)
+        gsutil.upload_text(
+            paths.pr_build_link,
+            paths.pr_path,
+            {'x-goog-meta-link': paths.pr_path}
```
Comment (rmmh) [inline jenkins/bootstrap.py]: `headers={...`

File: jenkins/bootstrap.py [hunk 4]
```diff
@@ -320,14 +324,18 @@ def copy_file(self, dest, orig):
...
+    def upload_text(self, path, txt, meta_links={}, cached=True):
         """Copy the text to path, optionally disabling caching."""
         headers = ['-h', 'Content-Type:text/plain']
         if not cached:
             headers += ['-h', 'Cache-Control:private, max-age=0, no-transform']
         cmd = [self.gsutil, '-q'] + headers + ['cp', '-', path]
         self.call(cmd, stdin=txt)
 
+        for meta, link in meta_links.iteritems():
+            cmd = [self.gsutil, 'setmeta', '-h', '\'%s: %s\'' % (meta, link), path]
```
Comment (rmmh) [inline jenkins/bootstrap.py]: this is unnecessary-- all headers can be set with the first cp command.

Comment (krzyzacy) [inline jenkins/bootstrap.py]: oh so `setmeta` is implied with `-h`?

Comment (rmmh) [inline jenkins/bootstrap.py]: gsutil cp writes metadata already, (Cache-Control etc are metadata), setmeta just modifies the metadata on an existing object.

**Comment by krzyzacy** [comment]
whoops... broke something there

**Pull request merged**